ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -40,25 +50,93 @@ private ListQueriesExecutor() { }
   ) {
     final ListQueries listQueries = statement.getStatement();
     if (listQueries.getShowExtended()) {
-      return Optional.of(new QueryDescriptionList(
-          statement.getStatementText(),
-          executionContext.getPersistentQueries().stream()
-              .map(QueryDescriptionFactory::forQueryMetadata)
-              .collect(Collectors.toList())));
+      return executeExtended(
+          listQueries,
+          statement,
+          sessionProperties,
+          executionContext,
+          serviceContext);
+    }
+
+    final List<RunningQuery> runningQueries = new ArrayList<>();
+    runningQueries.addAll(executionContext.getPersistentQueries()
+        .stream()
+        .map(q -> new RunningQuery(
+            q.getStatementString(),
+            ImmutableSet.of(q.getSinkName().name()),
+            ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
+            q.getQueryId(),
+            Optional.of(q.getState()),
+            Optional.of(new KsqlHostInfoEntity(sessionProperties.getKsqlHostInfo()))
+        ))
+        .collect(Collectors.toList()));
+    
+    if (!sessionProperties.getIsInternalRequest()) {
+      final Set<HostInfo> hosts =
+          DiscoverRemoteHostsUtil.getRemoteHosts(
+              executionContext.getPersistentQueries(),
+              sessionProperties.getKsqlHostInfo());
+
+      hosts.forEach(hostInfo -> {
+        final KsqlEntityList response = serviceContext.getKsqlClient().makeInternalKsqlRequest(
+            ServerUtil.buildRemoteUri(
+                sessionProperties.getLocalUrl(),
+                hostInfo.host(),
+                hostInfo.port()
+            ),
+            "show queries;"
```

Comment:
Avoid hardcoding SQL into the code.  The statement text is already available in `statement.getStatementText()`.